### PR TITLE
r/aws_schemas_registry_policy: new resource

### LINKED ISSUE #20831: Support Amazon EventBridge Schema Registry Resource-based policy
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon EventBridge Schemas Registry Resource-based policies are not currently supported by terraform.

[Here's the matching cloudformation resource: AWS::EventSchemas::RegistryPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eventschemas-registrypolicy.html)

### New or Affected Resource(s)

* aws_schemas_registry_policy

### References

- https://github.com/hashicorp/terraform-provider-aws/issues/11088

* [AWS::EventSchemas::RegistryPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eventschemas-registrypolicy.html)

### PR DESCRIPTION
**Pull request by georgealton**
### Description
Adds a new resource `aws_schemas_registry_policy` for creating AWS EventBridge Schema Registry Resource Policies.

### Relations

Closes #20831

### References

https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-resource-based-schemas.html

### Output from Acceptance Testing

```
make testacc TF_DEBUG=1 TESTS='TestAccSchemasRegistryPolicy' PKG=schemas
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/schemas/... -v -count 1 -parallel 20 -run='TestAccSchemasRegistryPolicy'  -timeout 180m
=== RUN   TestAccSchemasRegistryPolicy_basic
=== PAUSE TestAccSchemasRegistryPolicy_basic
=== RUN   TestAccSchemasRegistryPolicy_disappears
=== PAUSE TestAccSchemasRegistryPolicy_disappears
=== RUN   TestAccSchemasRegistryPolicy_disappears_Registry
=== PAUSE TestAccSchemasRegistryPolicy_disappears_Registry
=== RUN   TestAccSchemasRegistryPolicy_Policy
=== PAUSE TestAccSchemasRegistryPolicy_Policy
=== CONT  TestAccSchemasRegistryPolicy_basic
=== CONT  TestAccSchemasRegistryPolicy_disappears_Registry
=== CONT  TestAccSchemasRegistryPolicy_Policy
=== CONT  TestAccSchemasRegistryPolicy_disappears
--- PASS: TestAccSchemasRegistryPolicy_disappears (71.13s)
--- PASS: TestAccSchemasRegistryPolicy_disappears_Registry (73.10s)
--- PASS: TestAccSchemasRegistryPolicy_basic (83.27s)
--- PASS: TestAccSchemasRegistryPolicy_Policy (113.24s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/schemas	117.786s
...
```

### CODE REVIEW

File: website/docs/r/schemas_registry_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    principals {
+      type = "AWS"
+      identifiers = [
+        "109876543210"
+      ]
+    }
+    actions = ["schemas:*"]
+    resources = [
+      "arn:aws:schemas:us-east-1:012345678901:registry/default",
+      "arn:aws:schemas:us-east-1:012345678901:schema/default*"
```
Comment (georgealton) [inline website/docs/r/schemas_registry_policy.html.markdown:31]:
```suggestion
      "arn:aws:schemas:us-east-1:012345678901:registry/example",
      "arn:aws:schemas:us-east-1:012345678901:schema/example*"
```

Comment (georgealton) [reply website/docs/r/schemas_registry_policy.html.markdown:31]: Fix example

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccSchemasRegistryPolicy_' PKG=schemas ACCTEST_PARALLELISM=2 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/schemas/... -v -count 1 -parallel 2  -run=TestAccSchemasRegistryPolicy_ -timeout 180m
=== RUN   TestAccSchemasRegistryPolicy_basic
=== PAUSE TestAccSchemasRegistryPolicy_basic
=== RUN   TestAccSchemasRegistryPolicy_disappears
=== PAUSE TestAccSchemasRegistryPolicy_disappears
=== RUN   TestAccSchemasRegistryPolicy_disappears_Registry
=== PAUSE TestAccSchemasRegistryPolicy_disappears_Registry
=== RUN   TestAccSchemasRegistryPolicy_Policy
=== PAUSE TestAccSchemasRegistryPolicy_Policy
=== CONT  TestAccSchemasRegistryPolicy_basic
=== CONT  TestAccSchemasRegistryPolicy_disappears_Registry
--- PASS: TestAccSchemasRegistryPolicy_disappears_Registry (22.62s)
=== CONT  TestAccSchemasRegistryPolicy_disappears
--- PASS: TestAccSchemasRegistryPolicy_basic (28.42s)
=== CONT  TestAccSchemasRegistryPolicy_Policy
--- PASS: TestAccSchemasRegistryPolicy_disappears (19.62s)
--- PASS: TestAccSchemasRegistryPolicy_Policy (38.29s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/schemas	73.356s
```

**Comment by ewbankkit** [comment]
@georgealton Thanks for the contribution :tada: :clap:.

**Pull request merged**